TLM/TLM/Manager/Impl/UtilityManager.cs

```diff
@@ -18,26 +20,38 @@ public class UtilityManager : AbstractCustomManager, IUtilityManager {
 
         public static UtilityManager Instance { get; }
 
-        private readonly Queue<KeyValuePair<IRecordable, Dictionary<InstanceID, InstanceID>>> _transferRecordables = new ();
+        private readonly Queue<KeyValuePair<IRecordable, Dictionary<InstanceID, InstanceID>>> _transferRecordables = new();
 
-        public void ClearTraffic() {
+        public static void DespawnVehicles(ExtVehicleType? filter = null) {
             lock (Singleton<VehicleManager>.instance) {
                 try {
-                    VehicleManager vehicleManager = Singleton<VehicleManager>.instance;
-                    for (uint i = 0; i < vehicleManager.m_vehicles.m_size; ++i) {
-                        if ((vehicleManager.m_vehicles.m_buffer[i].m_flags &
-                             Vehicle.Flags.Created) != 0) {
-                            vehicleManager.ReleaseVehicle((ushort)i);
+                    var manager = Singleton<VehicleManager>.instance;
+
+                    for (uint vehicleId = 0; vehicleId < manager.m_vehicles.m_size; ++vehicleId) {
+                        ref Vehicle vehicle = ref ((ushort)vehicleId).ToVehicle();
+
+                        if (!vehicle.IsValid()) {
+                            continue;
+                        }
+
+                        if (filter != null &&
```

Comment:
`filter.HasValue`? It's `Nullable<ExtVehicleType>` so that should work too (haven't tried).

Reply:
Changed to `HasValue`